mlir/test/python/dialects/ods_helpers.py

```diff
@@ -208,3 +209,70 @@ class TestOp(OpView):
 
 
 run(testOdsBuildDefaultCastError)
+
+
+def testOdsEquallySizedAccessor():
+    class TestOpMultiResultSegments(OpView):
+        OPERATION_NAME = "custom.test_op"
+        _ODS_REGIONS = (1, True)
```

Comment:
Sorry just wondering, is there a reason you did this instead of just using `SameVariadicResultSizeOp`? Like I guess the tests you have here serve as strict "unit test" tests (because they directly hit `equally_sized_accessor`) but wouldn't exercising through `SameVariadicResultSizeOp` be the real test that the change is correct? I.e., doing something like

```python
op = SameVariadicResultSize(ts[0], [ts[1], ts[2], ts[3]], [ts[4], ts[5], ts[6]])
print(op.non_variadic)
print(op.variadic1)
print(op.variadic2)
```

Again, sorry if I'm missing some nuance here.

Reply:
I didn't consider it, but I think it is a good idea.
It seems like `SameVariadicResultSize` (and `SameVariadicOperandSize`) are traits defined in `OpBase.td`. However, from what I saw from quickly inspecting `mlir/lib/Bindings/Python`, I do not think the traits are currently exposed through the Python API.
Am I missing something?

Reply:
I goofed and misread what `mlir/test/mlir-tblgen/op-python-bindings.td` is - it is not the test that actually generates python bindings that can be used in tests. That's [mlir/test/python/python_test_ops.td](https://github.com/llvm/llvm-project/blob/9315645834ea81cf9550364a4950f289e9706a26/mlir/test/python/python_test_ops.td). 

> I do not think the traits are currently exposed through the Python API.

What I'm proposing is you just copy-paste 

```
def SameVariadicResultSizeOp : TestOp<"same_variadic_result",
                                      [SameVariadicResultSize]> {
  let results = (outs Variadic<AnyType>:$variadic1, AnyType:$non_variadic,
                 Variadic<AnyType>:$variadic2);
}
```

into `mlir/test/python/python_test_ops.td` and then create/query it in `python_test.py`.

Reply:
Very good call with the proposed extra test.
I've added it now, and in the process figured out that more information was needed in the `equally_sized_accessor` function.
I did a bit of refactoring in the `OpPythonBindingGen.cpp` file, so the `emitElementAccessors` does not need to take in as many function pointers.